Mathlib/FieldTheory/JacobsonNoether.lean

```diff
@@ -0,0 +1,198 @@
+  | succ _ h => rw [Function.iterate_succ', Function.comp_apply, h,
+    pow_add, pow_one, mulRight_apply, mul_assoc]
+
+private lemma δ_iterate_succ (x y : D) (n : ℕ) :
+    δ x (((δ x) ^ n) y) = ((δ x) ^ (n + 1)) y := by
+  simp only [pow_apply, δ_def, mulLeft_apply, mulRight_apply, Function.iterate_succ_apply']
+
+variable [Algebra.IsAlgebraic (Subring.center D) D]
+
+open Polynomial
+
+/-- If `D` is a purely inseparable extension of `k` with characteristic `p`,
+  then for every element `a` of `D`, there exists a natural number `n`
+  such that `a ^ (p ^ n)` is contained in `k`. -/
+lemma exists_pow_mem_center_of_inseparable (p : ℕ) [hchar : ExpChar D p] (a : D)
+    (hinsep : ∀ x : D, IsSeparable k x → x ∈ k) : ∃ n, a ^ (p ^ n) ∈ k := by
+  have := (@isPurelyInseparable_iff_pow_mem k D _ _ _ _ p (ExpChar.center_expChar_iff.1 hchar)).1
+  have pure : IsPurelyInseparable k D := ⟨Algebra.IsAlgebraic.isIntegral, fun x hx ↦ by
+    rw [RingHom.mem_range, Subtype.exists]
+    exact ⟨x, ⟨(hinsep x hx), rfl⟩⟩⟩
+  obtain ⟨n, ⟨m, hm⟩⟩ := (this pure) a
+  have := Subalgebra.range_subset (R := k) ⟨(k).toSubsemiring, by intro r; exact r.2⟩
+  exact ⟨n, Set.mem_of_subset_of_mem this <| Set.mem_range.2 ⟨m, hm⟩⟩
+
+/-- If `D` is a purely inseparable extension of `k` with characteristic `p`,
+  then for every element `a` of `D \ k`, there exists a natural number `n`
+  **greater than 0** such that `a ^ (p ^ n)` is contained in `k`. -/
+lemma exists_pow_mem_center_of_inseparable' (p : ℕ) [ExpChar D p] {a : D}
+    (ha : a ∉ k) (hinsep : ∀ x : D, IsSeparable k x → x ∈ k) : ∃ n ≥ 1, a ^ (p ^ n) ∈ k := by
+  obtain ⟨n, hn⟩ := exists_pow_mem_center_of_inseparable p a hinsep
+  by_cases nzero : n = 0
+  · rw [nzero, pow_zero, pow_one] at hn
+    exact (ha hn).elim
+  · exact ⟨n, ⟨by omega, hn⟩⟩
+
+/-- If `D` is a purely inseparable extension of `k` of characteristic `p`,
+  then for every element `a` of `D \ k`, there exists a natural number `m`
+  greater than 0 such that `(a * x - x * a) ^ n = 0` (as linear maps) for
+  every `n` greater than `(p ^ m)`. -/
+lemma exist_pow_eq_zero_of_le (p : ℕ) [hchar : ExpChar D p]
+    {a : D} (ha : a ∉ k) (hinsep : ∀ x : D, IsSeparable k x → x ∈ k):
+  ∃ m ≥ 1, ∀ n ≥ (p ^ m), (δ a) ^ n = 0 := by
+  obtain ⟨m, hm⟩ := exists_pow_mem_center_of_inseparable' p ha hinsep
+  refine ⟨m, ⟨hm.1, fun n hn ↦ ?_⟩⟩
+  have inter : (δ a) ^ (p ^ m) = 0 := by
+    ext x
+    rw [δ, Pi.sub_apply, sub_pow_expChar_pow_of_commute p m (commute_mulLeft_right a a)]
+    rw [sub_apply, l_pow, r_pow, sub_eq_zero_of_eq]; rfl
+    suffices h : a ^ (p ^ m) ∈ k from (Subring.mem_center_iff.1 h x).symm
+    exact hm.2
+  rw [((Nat.sub_eq_iff_eq_add hn).1 rfl), pow_add, inter, mul_zero]
+
+variable (D) in
+/-- For a non-commutative finite dimensional division algebra D (with base ring being its center),
+  there exist an element `x` that is separable over its center-/
+theorem Jacobson_Noether (H : k ≠ (⊤ : Subring D)) :
```

Comment:
Sure, that would definitely be more explict.

By the way, apologize if I seems silly but why is the annotation talking about ```finite dimensional division algebra D``` instead of something like ```algebraic division algebra D```? Is finite equivalent to algebraic in this case?